ion-schema/src/violation.rs

```diff
@@ -86,12 +86,37 @@ impl Violation {
     }
 }
 
-// TODO: Implement Violation with proper indentation for the nested tree of violations
 impl fmt::Display for Violation {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "A validation error occurred: {}", self.message)?;
-        for v in &self.violations {
-            write!(f, "  {v}")?;
+        f.write_str(self.message.as_str())?;
+
+        let mut stack = vec![];
+        let mut violations_iter = self.violations.iter();
+        let mut violation = violations_iter.next();
+
+        let one_indent = "  ";
+        // TODO: Consider extracting this into a static or thread local variable.
+        let mut indentation = vec![one_indent.to_string()];
```

Comment:
This works, but why have a `Vec` of indentation `String`s instead of a single `String` that you add spacing to as the stack gets deeper?

Reply:
Lol, it's because I've been using too much Java recently, and I forgot that strings can be mutated in Rust. I'll fix it.